pkg/types/gcp/machinepools.go

```diff
@@ -56,4 +61,75 @@ func (a *MachinePool) Set(required *MachinePool) {
 	if required.OSDisk.DiskType != "" {
 		a.OSDisk.DiskType = required.OSDisk.DiskType
 	}
+
+	a.EncryptionKey.Set(required.EncryptionKey)
```

Comment:
The logic is not correct here for the case where `required.EncryptionKey` is not nil but `a.EncryptionKey` is nil. In that case, the result should be that `a.EncryptionKey` is not nil and set with the values from `required.EncryptionKey`. Instead, the result is that `a.EncryptionKey` is left as nil.

The same goes for setting `KMSKey` later.